lib/gyp/platform/win.js

```diff
@@ -409,67 +392,164 @@ win.getOSBits = function getOSBits() {
   const env = process.env;
-  let tools;
+  let version = win.getMSVSVersion();
 
+  let tools;
   // Try to find a MSVS installation
   if (version === 'auto' && env['VS140COMNTOOLS'] || version === '2015') {
     version = '2015';
-    tools =  path.join(env.VS140COMNTOOLS, '..', '..');
+    tools = path.join(env.VS140COMNTOOLS, '..', '..');
   }
   if (version === 'auto' && env['VS120COMNTOOLS'] || version === '2013') {
     version = '2013';
-    tools =  path.join(env.VS120COMNTOOLS, '..', '..');
+    tools = path.join(env.VS120COMNTOOLS, '..', '..');
   }
   // TODO(indutny): more versions?
   if (version === 'auto' && env['VS100COMNTOOLS'] || version === '2010') {
     version = '2010';
-    tools =  path.join(env.VS120COMNTOOLS, '..', '..');
+    tools = path.join(env.VS120COMNTOOLS, '..', '..');
   }
   // TODO(indutny): does it work with MSVS Express?
 
   if (version === 'auto') {
-    gyp.bindings.error('No Visual Studio found. When building - please ' +
-                       'run `ninja` from the MSVS console');
-    return;
+    throw new Error('No Visual Studio found. When building - please ' +
+      'run `ninja` from the MSVS console');
   }
 
-  // NOTE: Largely inspired by MSVSVersion.py
-  const bits = win.getOSBits();
-
-  let vcvars;
+  let vcEnvCmd;
   // TODO(indutny): proper escape for the .bat file
-  if (arch === 'ia32') {
-    if (bits === 64)
-      vcvars = '"' + path.join(tools, 'VC', 'vcvarsall.bat') + '" amd64_x86';
+  if (target_arch === 'ia32') {
+    if (hostBits === 64)
+      vcEnvCmd = '"' + path.join(tools, 'VC', 'vcvarsall.bat') + '" amd64_x86';
     else
-      vcvars = '"' + path.join(tools, 'Common7', 'Tools', 'vsvars32.bat') + '"';
-  } else if (arch === 'x64') {
+      vcEnvCmd = '"'
+        + path.join(tools, 'Common7', 'Tools', 'vsvars32.bat')
+        + '"';
+  } else if (target_arch === 'x64') {
     let arg;
-    if (bits === 64)
+    if (hostBits === 64)
       arg = 'amd64';
     else
       arg = 'x86_amd64';
-    vcvars = '"' + path.join(tools, 'VC', 'vcvarsall.bat') + '" ' + arg;
+    vcEnvCmd = '"' + path.join(tools, 'VC', 'vcvarsall.bat') + '" ' + arg;
   } else {
-    throw new Error(`Arch: '${arch}' is not supported on windows`);
+    throw new Error(`Arch: '${target_arch}' is not supported on windows`);
   }
+  return vcEnvCmd;
+}
+
+win.resolveDevEnvironment = function resolveDevEnvironment(target_arch) {
+  const hostBits = win.getOSBits();
 
-  let lines;
+  const vcEnvCmd = tryVS7(hostBits, target_arch) ||
+    findOldVcVarsFile(hostBits, target_arch);
+  let lines = [];
   try {
-    lines = gyp.bindings.execSync(`${vcvars} & set`, { env: {} }).toString()
-        .split(/\r\n/g);
+    lines = gyp.bindings.execSync(`${vcEnvCmd} & set`, {env: {}}).toString()
+      .trim().split(/\r\n/g);
   } catch (e) {
     gyp.bindings.error(e.message);
-    return;
   }
+  const env = lines.reduce((s, l) => {
+    const kv = l.split('=');
+    s[kv[0]] = kv[1];
```

Comment:
We could do a `post` - `pre` delta calculation. I use some of these delta file my day2day dev environment.